src/wp-includes/block-template.php

```diff
@@ -235,16 +235,61 @@ function get_the_block_template_html() {
 		return;
 	}
 
+	// Record all post content strings into this list to avoid double parsing.
+	$parsed_post_content = array();
+	add_filter(
+		'render_block_core/post-content',
+		function( $block_content ) use ( &$parsed_post_content ) {
+			$parsed_post_content[] = $block_content;
+			return $block_content;
+		},
+		PHP_INT_MAX
+	);
+
 	$content = $wp_embed->run_shortcode( $_wp_current_template_content );
 	$content = $wp_embed->autoembed( $content );
 	$content = do_blocks( $content );
```

Comment:
Could we do this?
```
remove_filter( 'the_excerpt', 'wp_filter_content_tags' );
remove_filter( 'the_content', 'wp_filter_content_tags' );
$content = do_blocks( $content );
add_filter( 'the_excerpt', 'wp_filter_content_tags' );
add_filter( 'the_content', 'wp_filter_content_tags' );
```

Reply:
He did not seem at all interested in assisting with a solution based on filter removal, even after I informed him that his implementation would break shortcodes. I just don't have the familiarity with `wp_filter_content_tags` to know how to implement it correctly, and all I can tell is that the `$context` variable needs to be set to `'the_content'` for it to work as intended.

Reply:
As noted in my PR, even after applying this PR, filters are applied double, as image filter is applied in template_part funciton.

Reply:
I'm not talking about the same thing. It's not because of the template part, it's because [there are conditions that require the the `$context` variable to be set to `'the_content'`](https://github.com/WordPress/wordpress-develop/blob/6.1/src/wp-includes/media.php#L5435) rather than `'the_block_template_html'`, and I'm not sure of the ramifications of changing the condition to allow a different value or explicitly passing `'the_content'` to make the condition pass. If @felixarntz is willing to assist in this area then we might be able to use https://github.com/WordPress/gutenberg/pull/44995 as the solution instead of putting up with his messy implementation.

Reply:
> Could we do this?

@spacedmonkey Removing the filters around the `do_blocks()` call is only a partial solution. This "record before, reinstate after" approach is the only one that solves _both_ "double execution problem" and _also_ the "avoid all execution problem" as discussed at length in [55996](https://core.trac.wordpress.org/ticket/55996).

If I need to open another ticket on Trac so that the "avoid all execution problem" is considered of equal importance to the "double execution problem" then I guess I'll do that :) In the meantime, this approach is the one that works for both problems.